src/app/core/services/authentication.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import {
+  BehaviorSubject,
+  catchError,
+  from,
+  map,
+  Observable,
+  tap,
+  throwError,
+} from 'rxjs';
+import firebase from 'firebase/compat';
+import FirebaseError = firebase.FirebaseError;
+import { UsersService } from './users.service';
+import { Store } from '@ngrx/store';
+import { removeToken, setToken } from '../states/auth/auth.actions';
+import { resolveError, setErrorMessage } from '../states/error/error.actions';
+import UserCredential = firebase.auth.UserCredential;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticationService {
+  loggedIn = new BehaviorSubject<boolean>(false);
+  loggedIn$ = this.loggedIn.asObservable();
+  constructor(
+    private auth: AngularFireAuth,
+    private usersService: UsersService,
+    private store: Store,
+  ) {
+    this.auth.onAuthStateChanged((user) => {
+      if (user) {
+        this.loggedIn.next(true);
+      } else {
+        this.loggedIn.next(false);
+      }
+    });
+  }
+
+  signInWithEmailPassword(
+    email: string,
+    password: string,
+  ): Observable<UserCredential> {
+    return from(this.auth.signInWithEmailAndPassword(email, password)).pipe(
+      tap(
+        (result) =>
+          result.user
+            ?.getIdToken()
+            .then((token) => this.store.dispatch(setToken({ token: token }))),
+      ),
+      tap(() => this.store.dispatch(resolveError({ errorType: 'signIn' }))),
```

Comment:
Is that clearing any previous error? That should semantically come before the request or the user waits for the network until it's cleared?